From d40eff53e2aa8897f5df2b9549761cf7af322085 Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Thu, 11 Jan 2018 12:33:29 +0500
Subject: [PATCH] configure.ac: for llvm-config to report correct libdir

In cross compiling environments llvm-config messes up
as it reports native environment because it is built
for the host system. Hence if the target system and
host system have different baselibs the llvm-config
fails to find libraries for the target system
appropriately.
This now forces llvm-config to use the target specific
libdir.

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 configure.ac | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 2561956..f7f65d6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -998,7 +998,7 @@ llvm_set_environment_variables() {
         LLVM_VERSION=`$LLVM_CONFIG --version | cut -c1-5`
         LLVM_CPPFLAGS=`strip_unwanted_llvm_flags "$LLVM_CONFIG --cppflags"`
         LLVM_INCLUDEDIR=`$LLVM_CONFIG --includedir`
-        LLVM_LIBDIR=`$LLVM_CONFIG --libdir`
+        LLVM_LIBDIR=${llvm_prefix}
 
         # We need to respect LLVM_CPPFLAGS when compiling LLVM headers.
         save_CFLAGS="$CFLAGS"
@@ -2684,17 +2684,17 @@ if test "x$enable_llvm" = xyes; then
 
     if test $LLVM_VERSION_MAJOR -ge 4 -o $LLVM_VERSION_MAJOR -eq 3 -a $LLVM_VERSION_MINOR -ge 9; then
         if test "x$enable_llvm_shared_libs" = xyes; then
-            LLVM_LIBS="`$LLVM_CONFIG --link-shared --libs ${LLVM_COMPONENTS}`"
+            LLVM_LIBS="`$LLVM_CONFIG --link-shared --tgtlibdir ${llvm_prefix} --libs ${LLVM_COMPONENTS}`"
         else
             dnl Invoking llvm-config with both -libs and --system-libs produces the
             dnl two separate lines - each for the set of libraries.
 		dnl Call the program twice, effectively folding them into a single line.
-            LLVM_LIBS="`$LLVM_CONFIG --link-static --libs ${LLVM_COMPONENTS}`"
+            LLVM_LIBS="`$LLVM_CONFIG --link-static --tgtlibdir ${llvm_prefix} --libs ${LLVM_COMPONENTS}`"
             dnl We need to link to llvm system libs when using static libs
             LLVM_LIBS="$LLVM_LIBS `$LLVM_CONFIG --link-static --system-libs`"
         fi
     else
-        LLVM_LIBS="`$LLVM_CONFIG --libs ${LLVM_COMPONENTS}`"
+        LLVM_LIBS="`$LLVM_CONFIG --tgtlibdir ${llvm_prefix} --libs ${LLVM_COMPONENTS}`"
         if test "x$enable_llvm_shared_libs" = xyes; then
             detect_old_buggy_llvm
         else
-- 
2.11.1

